ttnn/cpp/ttnn/operations/full/device/full_device_operation.cpp

```diff
@@ -0,0 +1,90 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "full_device_operation.hpp"
+
+#include "ttnn/tensor/tensor.hpp"
+
+namespace ttnn::operations::full {
+void FullOperation::validate_inputs(
+    const operation_attributes_t& operation_attributes, const tensor_args_t& tensor_args) {
+    auto any = tensor_args.any;
+    TT_FATAL(any.storage_type() == StorageType::DEVICE, "Full operation error: Any tensor must be on device");
+    TT_FATAL(
+        operation_attributes.memory_config.memory_layout == TensorMemoryLayout::INTERLEAVED,
+        "Full operation error: Not currently supporting sharding");
+    TT_FATAL(
+        operation_attributes.layout == Layout::TILE, "Full operation error: Not currently supporting row major layout");
+
+    const auto shape = operation_attributes.shape;
+
+    TT_FATAL(
+        shape.size() > 1,
+        "Full operation error: Shape size must be greater than 1, but got shape size = {}",
+        shape.size());
+
+    for (int i = 0; i < shape.size(); i++) {
```

Comment:
If this shape is SimpleShape (and it should ideally be that) - you can check shape's volume against 0.

Altough if you make it SimpleShape - keep in mind that type is not yet bound to python.

Reply:
Since `SimpleShape` is not bound to Python, I’ve kept it as a vector